client/a8c-for-agencies/sections/partner-directory/partner-directory.tsx

```diff
@@ -0,0 +1,91 @@
+import { useTranslate } from 'i18n-calypso';
+import { ReactNode, useMemo } from 'react';
+import Layout from 'calypso/a8c-for-agencies/components/layout';
+import LayoutBody from 'calypso/a8c-for-agencies/components/layout/body';
+import LayoutHeader, {
+	LayoutHeaderBreadcrumb as Breadcrumb,
+	LayoutHeaderTitle as Title,
+} from 'calypso/a8c-for-agencies/components/layout/header';
+import LayoutTop from 'calypso/a8c-for-agencies/components/layout/top';
+import MobileSidebarNavigation from 'calypso/a8c-for-agencies/components/sidebar/mobile-sidebar-navigation';
+import { A4A_PARTNER_DIRECTORY_LINK } from 'calypso/a8c-for-agencies/components/sidebar-menu/lib/constants';
+import { Item as BreadcrumbItem } from 'calypso/components/breadcrumb';
+import AgencyDetails from './agency-details';
+import AgencyExpertise from './agency-expertise';
+import {
+	PARTNER_DIRECTORY_AGENCY_DETAILS_SLUG,
+	PARTNER_DIRECTORY_AGENCY_EXPERTISE_SLUG,
+	PARTNER_DIRECTORY_DASHBOARD_SLUG,
+} from './constants';
+import Dashboard from './dashboard';
+
+type Props = {
+	selectedSection: string;
+};
+
+interface Section {
+	content: ReactNode;
+	breadcrumbItems: BreadcrumbItem[];
+}
+
+export default function PartnerDirectory( { selectedSection }: Props ) {
+	const translate = useTranslate();
+	const title = translate( 'Partner Directory' );
+
+	// Define the sub-menu sections
+	const sections: { [ slug: string ]: Section } = useMemo( () => {
+		const sections: { [ slug: string ]: Section } = {};
+
+		sections[ PARTNER_DIRECTORY_DASHBOARD_SLUG ] = {
+			content: <Dashboard />,
+			breadcrumbItems: [
+				{
+					label: translate( 'Partner Directory' ),
+					href: A4A_PARTNER_DIRECTORY_LINK,
+				},
+			],
+		};
+
+		sections[ PARTNER_DIRECTORY_AGENCY_DETAILS_SLUG ] = {
+			content: <AgencyDetails />,
+			breadcrumbItems: [
+				...sections[ PARTNER_DIRECTORY_DASHBOARD_SLUG ].breadcrumbItems,
+				{
+					label: translate( 'Agency Details' ),
+					href: `${ A4A_PARTNER_DIRECTORY_LINK }/${ PARTNER_DIRECTORY_AGENCY_DETAILS_SLUG }`,
+				},
+			],
+		};
+
+		sections[ PARTNER_DIRECTORY_AGENCY_EXPERTISE_SLUG ] = {
+			content: <AgencyExpertise />,
+			breadcrumbItems: [
+				...sections[ PARTNER_DIRECTORY_AGENCY_DETAILS_SLUG ].breadcrumbItems,
+				{
+					label: translate( 'Agency Expertise' ),
+					href: `${ A4A_PARTNER_DIRECTORY_LINK }/${ PARTNER_DIRECTORY_AGENCY_EXPERTISE_SLUG }`,
+				},
+			],
+		};
+
+		return sections;
+	}, [ translate ] );
```

Comment:
> The only potential downside I foresee is that this may become complex as each section might evolve and deviate from the common design. (For instance, sticky headers, unique CTAs, etc.)

If so, we can see those edge cases in the meetup. If you are referring to the layout refactor, we have a way to make it customizable. For example, we could attach the section object to the section component; that way, we could update the upper layer (the Layout) from the section component. It would be a customization from the child component to the parent.

The idea of the refactor is to have the central section layout, and then, it's just change the section content of each section instead of re-rendering the layout (duplicated code)